js/blocks/controls/fetch-input.js

```diff
@@ -0,0 +1,304 @@
+const { Component, createRef } = wp.element;
+const { decodeEntities } = wp.htmlEntities;
+const { UP, DOWN, ENTER, TAB } = wp.keycodes;
+const { BaseControl, Spinner, withSpokenMessages, Popover } = wp.components;
+const { withInstanceId } = wp.compose;
+const apiFetch = wp.apiFetch;
+const { addQueryArgs } = wp.url;
+
+// Since FetchInput is rendered in the context of other inputs, but should be
+// considered a separate modal node, prevent keyboard events from propagating
+// as being considered from the input.
+const stopEventPropagation = ( event ) => event.stopPropagation();
+
+class FetchInput extends Component {
+	constructor( { autocompleteRef } ) {
+		super( ...arguments );
+
+		this.onBlur = this.onBlur.bind( this );
+		this.onChange = this.onChange.bind( this );
+		this.onKeyDown = this.onKeyDown.bind( this );
+		this.autocompleteRef = autocompleteRef || createRef();
+		this.inputRef = createRef();
+		this.updateSuggestions = this.updateSuggestions.bind( this );
+
+		this.suggestionNodes = [];
+
+		this.state = {
+			results: [],
+			showSuggestions: false,
+			selectedSuggestion: null,
+		};
+	}
+
+	componentWillUnmount() {
+		delete this.suggestionsRequest;
+	}
+
+	bindSuggestionNode( index ) {
+		return ( ref ) => {
+			this.suggestionNodes[ index ] = ref;
+		};
+	}
+
+	updateSuggestions( value ) {
+		// Show the suggestions after typing at least 2 characters
+		// and also for URLs.
+		if ( value.length < 2 ) {
+			this.setState( {
+				showSuggestions: false,
+				selectedSuggestion: null,
+				loading: false,
+			} );
+
+			return;
+		}
+
+		this.setState( {
+			showSuggestions: true,
+			selectedSuggestion: null,
+			loading: true,
+		} );
+
+		const request = apiFetch( {
+			path: addQueryArgs( '/wp/v2/' + this.props.apiSlug, {
+				search: value,
+				per_page: 20,
+			} ),
+		} );
+
+		request.then( ( results ) => {
+			// A fetch Promise doesn't have an abort option. It's mimicked by
+			// comparing the request reference in on the instance, which is
+			// reset or deleted on subsequent requests or unmounting.
+			if ( this.suggestionsRequest !== request ) {
+				return;
+			}
+
+			this.setState( {
+				results,
+				loading: false,
+			} );
+
+			if ( !! results.length ) {
+				this.props.debouncedSpeak( sprintf( _n(
+					'%d result found, use up and down arrow keys to navigate.',
+					'%d results found, use up and down arrow keys to navigate.',
+					results.length,
+					'block-lab'
+				), results.length ), 'assertive' );
+			} else {
+				this.props.debouncedSpeak( __( 'No results.', 'block-lab' ), 'assertive' );
+			}
+		} ).catch( () => {
+			if ( this.suggestionsRequest === request ) {
+				this.setState( {
+					loading: false,
+				} );
+			}
+		} );
+
+		this.suggestionsRequest = request;
+	}
+
+	/**
+	 * On clicking the block outside the <input>, hide the Popover.
+	 * Mainly taken from the color control onBlur handler.
+	 */
+	onBlur( event ) {
+		console.log( event.relatedTarget.classList );
```

Comment:
Is this debug code?

Reply:
Ah, I should have removed this. https://github.com/getblocklab/block-lab/pull/233/commits/f53ba5cf764dadeb84554803502ebc2d94e08a58 deletes it.